bin/merge_md_files.rb

```
@@ -1,45 +1,146 @@
+                          .map(&:to_s)
+                          .reject { |f| %w[1page.md ai-assistant.md].include?(f) }
+                          .sort
+    bp_md_files = source.join("bp").directory? ?
+                  source.join("bp").glob("*.md").map(&:basename).map(&:to_s).sort : []
+    file_order = main_md_files + bp_md_files.map { |f| "bp/#{f}" }
+  end
+
+  # Gather all files: main files in order + best practices
+  all_files = file_order.map { |fn| source.join(fn) }.select(&:file?)
+  main_files = all_files.reject { |f| f.dirname.basename.to_s == "bp" }
+  bp_files = all_files.select { |f| f.dirname.basename.to_s == "bp" }
```

RuboCop (1.75.5)

[warning] Useless assignment to variable - `bp_files`. Did you mean `bp_md_files`?

(Lint/UselessAssignment)